UIView+Autolayout.swift

```diff
@@ -230,3 +230,50 @@ extension UIView {
     }
     
 }
+
+// MARK: Center superview
+extension UIView {
+    
+    @discardableResult func centerYToSuperview(withConstant constant: CGFloat = 0,
+                                               priority: UILayoutPriority = UILayoutPriorityRequired,
+                                               relatedBy relation: NSLayoutRelation = .equal) -> NSLayoutConstraint {
+        guard let superview = self.superview else {
+            preconditionFailure("view has no superview")
+        }
+        
+        translatesAutoresizingMaskIntoConstraints = false
+        
+        let constraint = NSLayoutConstraint(item: self,
+                                            attribute: .centerY,
+                                            relatedBy: relation,
+                                            toItem: superview,
+                                            attribute: .centerY,
+                                            multiplier: 1,
+                                            constant: constant)
```

Comment:
doesn't really matter from an internal perspective. We should probably _activate_ the constraint rather than adding it to the superview, but that's for another time